### PR TITLE
Simplify NewNodeID parameters

### PR DESCRIPTION
**Pull request by tim-ywliu**
Checking nodeID Type in method.

### CODE REVIEW

**Review by wmnsk — COMMENTED** [review]
Thanks!
I don’t want a breaking change as long as the current version is somehow functioning without bugs. Can you add it as `NewNodeIDHeuristic` instead? I’ll consider replacing the current one with it when we bump major version at some point.

**Comment by tim-ywliu** [comment]
@wmnsk 

Updated.

File: ie/node-id.go [hunk 1]
```diff
@@ -46,6 +46,33 @@ func NewNodeID(ipv4, ipv6, fqdn string) *IE {
...
+		return nil
+	}
+
+	var p []byte
+	ip := net.ParseIP(nodeID)
+	if ip != nil {
+		if ip.To4() != nil { // IPv4
+			p = make([]byte, 5)
+			p[0] = NodeIDIPv4Address
+			copy(p[1:], ip.To4())
```
Comment (wmnsk) [inline ie/node-id.go:61]:
```suggestion
		if v4 := ip.To4(); v4 != nil { // IPv4
			p = make([]byte, 5)
			p[0] = NodeIDIPv4Address
			copy(p[1:], v4)
```

You can avoid calling `To4()` twice (not sure if it really affect the performance - maybe compiler does something or maybe not :p)

**Review by wmnsk — APPROVED** [review]
Looks great, thanks!

**Pull request merged**